### ISSUE TITLE
False positive for F821: undefined name 'ignore'

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#352]
In GitLab by @sobolevn on Mar 19, 2020, 06:53

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
» flake8 --bug-report
{
  "dependencies": [
    {
      "dependency": "entrypoints",
      "version": "0.3"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.5",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-bandit",
      "version": "2.1.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-broken-line",
      "version": "0.2.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-bugbear",
      "version": "19.8.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-comprehensions",
      "version": "3.2.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-darglint",
      "version": "0.4.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-debugger",
      "version": "3.2.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-docstrings",
      "version": "1.5.0, pydocstyle: 5.0.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-eradicate",
      "version": "0.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-pyi",
      "version": "19.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-pytest-style",
      "version": "0.2.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-string-format",
      "version": "0.2.3"
    },
    {
      "is_local": false,
      "plugin": "flake8_builtins",
      "version": "1.5.2"
    },
    {
      "is_local": false,
      "plugin": "flake8_commas",
      "version": "2.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_isort",
      "version": "2.9.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_quotes",
      "version": "2.1.1"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "naming",
      "version": "0.9.1"
    },
    {
      "is_local": false,
      "plugin": "nitpick",
      "version": "0.21.4"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.5.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.1.1"
    },
    {
      "is_local": false,
      "plugin": "rst-docstrings",
      "version": "0.0.12"
    },
    {
      "is_local": false,
      "plugin": "wemake_python_styleguide",
      "version": "0.13.4"
    }
  ],
  "version": "3.7.9"
}

```

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

When I use this code sample:

```python
class Test(object):
    def write(  # type: ignore[override]
        self,
    ) -> None:
        ...
```

It raises these two violations:

```
  2:17     F821  undefined name 'ignore'
  def write(  # type: ignore[override]
              ^

  2:17     F821  undefined name 'override'
  def write(  # type: ignore[override]
              ^
```

Which is clearly not correct, because `ignore[<code>]` is a special syntax from `mypy` to disable some specific error types for a specific line.

Docs: https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker

I expect this code to be valid (in terms of `F821` check).

**Comment by asottile** [comment]
In GitLab by @sobolevn on Mar 19, 2020, 06:53

Related: https://github.com/wemake-services/wemake-python-styleguide/issues/1269

**Comment by asottile** [comment]
In GitLab by @sobolevn on Mar 19, 2020, 06:54

changed the description

**Comment by asottile** [comment]
In GitLab by @sobolevn on Mar 19, 2020, 07:00

Solved in https://github.com/PyCQA/pyflakes/issues/475 It is just not released yet.

**Comment by asottile** [comment]
In GitLab by @sobolevn on Mar 19, 2020, 07:00

closed

**Comment by asottile** [comment]
In GitLab by @asottile on Mar 19, 2020, 10:01

note that flake8 does not implement any checks of its own, please in the future open issues on the underlying projects
